File: exemplos_python/Aleatorio.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
Robot: Aleatorio

Movimenta-se aleatóriamente pelo campo de batalha.
"""
from port_module import Robot


class Aleatorio( Robot ):
    def __init__( self ):
        Robot.__init__( self )
        # Enquanto True o robô continua a funcionar
        self._alive = True
        # Controla a direcção do movimento do robô
        self._drection = 1
        # Inicia o robot
        self.init()
    


    def round_started( self, roundNum ):
        print "\nround_started( " + str( roundNum ) + ' )'
        self._direction = 1
        self.exec_mode("block")
        self.subscribe_events( "[on_hit_robot, on_hit_wall]" )
    
    
    
    def on_hit_wall( self, wall ):
        print "on_hit_wall( " + wall + " )"
        self._direction *= -1
    
    def on_hit_robot( self, event ):
        print "#################################"
        print "on_hit_robot()"
        print event.robotName
        print event.robotDamage
        print event.robotArmor
        print event.relativeDirection
        print "#################################"
        self._direction *= -1



    def start( self ):
        while( self._alive ):
            self.robot_move_forward( 300 * self._direction )
            self.robot_turn_right( 90 )
            self.execute()







if __name__ == '__main__':
    robot = Aleatorio()
    robot.start()

File: exemplos_java/Aleatorio.java
/*
 * Robot: Aleatorio
 * 
 * Movimenta-se aleatóriamente pelo campo de batalha.
 */

import java.io.*;
import java.util.*;


class Aleatorio extends Robot {
	
	public void round_started( int args ) {
		System.out.println( "round_started( " + args +  " )" );
		moving = 1;
        exec_mode("block");
        subscribe_events( "[on_hit_robot, on_hit_wall]" );
	}
	
	
	
	public void on_hit_wall( String wall ) {
		System.out.println( "on_hit_wall( " + wall + " )" );
		moving *= -1;
	}
	
	public void on_hit_robot( OnHitRobotEvent event ) {
		System.out.println( "#################################" );
		System.out.println( "on_hit_robot()" );
		System.out.println( event.robotName );
		System.out.println( event.robotDamage );
		System.out.println( event.robotArmor );
		System.out.println( event.relativeDirection );
		System.out.println( "#################################" );
		moving *= -1;
	}
	

	
	
	
	// Enquanto true o robô continua a funcionar
	boolean alive = true;
	// Controla a direcção do movimento do robô
	int moving = 1;	

	void start() {
		// Inicia o robô
		init();
        while ( alive ) {
	        robot_move_forward( 300 * moving );
	        robot_turn_right( 90 );
	        execute();
        }
	}
	
	
	
	
	
	

	public static void main( String[] args ) {
		Aleatorio robot = new Aleatorio();
		robot.start();
	}
	
}